wolfcrypt/src/curve25519.c

```diff
@@ -51,12 +51,45 @@ const curve25519_set_type curve25519_sets[] = {
     }
 };
 
+/* compute the public key from an existing private key, using bare vectors. */
+int wc_curve25519(int public_size, byte* public, int private_size, const byte* private) {
+    int ret;
+
+    if ((public_size != CURVE25519_KEYSIZE) ||
+        (private_size != CURVE25519_KEYSIZE))
+        return ECC_BAD_ARG_E;
+    if ((public == NULL) || (private == NULL))
+        return ECC_BAD_ARG_E;
+
+    /* check clamping */
+    if ((private[0] & ~248) ||
+        (private[CURVE25519_KEYSIZE-1] & 128))
+        return ECC_BAD_ARG_E;
+
+#ifdef FREESCALE_LTC_ECC
+    {
+        const ECPoint* basepoint = nxp_ltc_curve25519_GetBasePoint();
+        ECPoint wc_pub;
+        ret = nxp_ltc_curve25519(&wc_pub, private, basepoint, kLTC_Weierstrass); /* input basepoint on Weierstrass curve */
+        if (ret == 0)
+            XMEMCPY(public, wc_pub.point, CURVE25519_KEY_SIZE);
+    }
+#else
+    {
+        static const unsigned char basepoint[CURVE25519_KEYSIZE] = {9};
```

Comment:
Can we move this to global space and call `kCurveBasePoint` or something? Its duplicated a few places.

Reply:
I've split the difference and made it a top-level static in `curve25519.c` named `kCurve25519BasePoint` (reduces from 2 to 1 instance in the data segment).  This constant is used once more in `fe_operations.h` as `c25519_base_x` ironically `#if defined(CURVE25519_SMALL) || defined(ED25519_SMALL)`, but it may be getting optimized out, as it's `static` and nothing uses it.